app/services/post_scraper.rb

```diff
@@ -143,179 +109,80 @@ def threaded_page_links
+    comments = comments.first(25).compact if @threaded_import # can't do 25 on non-threaded because single page is 50 per
 
     comments.each do |comment|
       content = comment.at_css('.comment-content').inner_html
-      img_url = comment.at_css('.userpic img').try(:attribute, 'src').try(:value)
-      img_keyword = comment.at_css('.userpic img').try(:attribute, 'title').try(:value)
+      img_node = comment.at_css('.userpic img')
+      img_url = img_node.try(:attribute, 'src').try(:value)
+      img_keyword = img_node.try(:attribute, 'title').try(:value)
       username = comment.at_css('.comment-poster b').inner_html
       created_at = comment.at_css('.datetime').text
 
-      @reply = Reply.new
-      @reply.post = @post
-      @reply.content = strip_content(content)
-      @reply.created_at = @reply.updated_at = created_at
-
-      set_from_username(@reply, username)
-      set_from_icon(@reply, img_url, img_keyword)
+      reply = @post.replies.new(skip_notify: true, skip_post_update: true, skip_regenerate: true, is_import: true)
 
-      @reply.skip_notify = true
-      @reply.skip_post_update = true
-      @reply.skip_regenerate = true
-      @reply.is_import = true
-      Audited.audit_class.as_user(@reply.user) do
-        @reply.save!
-      end
+      scraper = ReplyScraper.new(reply, errors: @errors, console: @console_import)
+      scraper.import(username: username, img_url: img_url, img_keyword: img_keyword, content: content, created_at: created_at)
     end
   end
 
   def finalize_post_data
-    @post.last_user_id = @reply.try(:user_id) || @post.user_id
-    @post.last_reply_id = @reply.try(:id)
-    @post.tagged_at = @reply.try(:created_at) || @post.created_at
+    last_reply = @post.replies.last
+    @post.last_user_id = (last_reply || @post).user_id
+    @post.last_reply_id = last_reply.id if last_reply
+    @post.tagged_at = (last_reply || @post).created_at
     @post.authors_locked = true
     @post.save!
   end
 
-  def set_from_username(tag, username)
-    if BASE_ACCOUNTS.key?(username)
-      tag.user = User.find_by(username: BASE_ACCOUNTS[username])
-      return
-    end
-
-    character = Character.find_by(screenname: username.tr("-", "_")) || Character.find_by(screenname: username.tr("_", "-"))
-    unless character
-      user = prompt_for_user(username)
-      character = Character.create!(user: user, name: username, screenname: username)
-      gallery = Gallery.create!(user: user, name: username)
-      CharactersGallery.create!(character_id: character.id, gallery_id: gallery.id)
-    end
-
-    tag.character = character
-    tag.user = character.user
+  def clean_url(url)
+    uri = URI(url)
+    query = Rack::Utils.parse_query(uri.query)
+    return url if check_query(query)
+    query.except!('style')
+    query['view'] = 'flat' unless @threaded_import
+    new_query = Rack::Utils.build_query(query)
+    URI::HTTPS.build(host: uri.host, path: uri.path, fragment: uri.fragment, query: new_query).to_s
```

Comment:
Is there a need to specify everything here? I think we can just do `uri.query = query` and then `uri.to_s`.

Reply:
...I looked and looked for a way to do that but I didn't think of this, thank you.